src/types/block_certificates.rs

```diff
@@ -0,0 +1,120 @@
+//! Module exposing certificates for blocks.
+//! The types are only relevant for nodes running at least protocol version
+//! 6.
+
+use concordium_base::{
+    base::{BakerId, Epoch, Round},
+    common::Serial,
+    hashes::BlockHash,
+};
+
+/// An aggregate signature on a [`QuorumCertificate`] created
+/// by members of the finalization committee.
+#[derive(concordium_base::common::Serialize, Clone, Debug, PartialEq)]
+pub struct QuorumSignature(
+    pub concordium_base::aggregate_sig::Signature<concordium_base::base::AggregateSigPairing>,
+);
+
+/// A quorum certificate on a block.
+#[derive(Debug)]
+pub struct QuorumCertificate {
+    /// [`BlockHash`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub block_hash:          BlockHash,
+    /// [`Round`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub round:               Round,
+    /// [`Epoch`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub epoch:               Epoch,
+    /// The aggregate signature on the block identified by the
+    /// `block_hash` which serves as a proof that the block
+    /// was accepted on the chain by a quorum of the finalization
+    /// committee.
+    pub aggregate_signature: QuorumSignature,
+    /// The baker ids of the finalizers that formed
+    /// the `aggregate_signature`.
+    /// Note that the baker ids are sorted in ascending order and are
+    /// distinct.
+    pub signatories:         Vec<BakerId>,
```

Comment:
So is there a reason for not using `BTreeSet` here instead?

That would encode the property you have listed in the type.